stub.py

```diff
@@ -0,0 +1,94 @@
+"""Stub class"""
+
+from abc import abstractmethod
+from enum import Enum
+
+from pydantic.v1 import BaseModel
+
+
+class TaskType(str, Enum):
+    FDTD = "FDTD"
+    NORMAL = "NORMAL"
+    MODE_SOLVER = "MODE_SOLVER"
+    HEAT = "HEAT"
+
+
+class Stub(BaseModel):
+    @abstractmethod
+    def from_file(self, file_path: str) -> "Stub":
+        @abstractmethod
+        def from_file(self, file_path: str) -> "Stub":
+            """Loads a :class:`Stub` from .yaml, .json, or .hdf5 file.
+
+            Parameters
+            ----------
+            file_path : str
+                Full path to the .yaml or .json or .hdf5 file to load the :class:`Stub` from.
+
+            Returns
+            -------
+            :class:`Stub`
+                An instance of the component class calling `load`.
+
+            Example
+            -------
+            >>> simulation = Simulation.from_file(fname='folder/sim.json') # doctest: +SKIP
+            """
+            pass
+
+    @abstractmethod
+    def to_file(
+        self,
+        file_path: str,
+    ) -> None:
+        """Exports :class:`Stub` instance to .yaml, .json, or .hdf5 file
+
+        Parameters
+        ----------
+        file_path : str
+            Full path to the .yaml or .json or .hdf5 file to save the :class:`Stub` to.
+
+        Example
+        -------
+        >>> simulation.to_file(fname='folder/sim.json') # doctest: +SKIP
+        """
+        pass
+
+    @abstractmethod
+    def get_type(self) -> TaskType:
+        """Get Stub type.
+
+        Returns
+        -------
+        :class:`TaskType`
+            An instance Type of the component class calling `load`.
+        """
+        pass
+
+    @abstractmethod
```

Comment:
I dont think we need to enforce that this is implemented in subclasses. so maybe we can just remove `abstractmethod` but leave it as is.

Reply:
removed